DigitalLearningSolutions.Data/Services/CourseAdminFieldsService.cs

```diff
@@ -0,0 +1,217 @@
+
+    public interface ICourseAdminFieldsService
+    {
+        public CourseAdminFields? GetCustomPromptsForCourse(int customisationId, int centreId, int categoryId);
+
+        public List<CustomPromptWithAnswer> GetCustomPromptsWithAnswersForCourse(
+            DelegateCourseInfo delegateCourseInfo,
+            int customisationId,
+            int centreId,
+            int categoryId = 0
+        );
+
+        public void UpdateCustomPromptForCourse(int customisationId, int promptNumber, bool mandatory, string? options);
+
+        public void RemoveCustomPromptFromCourse(int customisationId, int promptNumber);
+
+        public string GetPromptNameForCourseAndPromptNumber(int customisationId, int promptNumber);
+    }
+
+    public class CourseAdminFieldsService : ICourseAdminFieldsService
+    {
+        private readonly ICourseAdminFieldsDataService courseAdminFieldsDataService;
+        private readonly ILogger<CourseAdminFieldsService> logger;
+        private readonly IUserDataService userDataService;
+
+        public CourseAdminFieldsService(
+            ICourseAdminFieldsDataService courseAdminFieldsDataService,
+            ILogger<CourseAdminFieldsService> logger,
+            IUserDataService userDataService
+        )
+        {
+            this.courseAdminFieldsDataService = courseAdminFieldsDataService;
+            this.logger = logger;
+            this.userDataService = userDataService;
+        }
+
+        public CourseAdminFields? GetCustomPromptsForCourse(
+            int customisationId,
+            int centreId,
+            int categoryId = 0
+        )
+        {
+            var result = courseAdminFieldsDataService.GetCourseAdminFields(customisationId, centreId, categoryId);
+            return new CourseAdminFields(
+                customisationId,
+                centreId,
+                PopulateCustomPromptListFromCourseCustomPromptsResult(result)
+            );
+        }
+
+        public List<CustomPromptWithAnswer> GetCustomPromptsWithAnswersForCourse(
+            DelegateCourseInfo delegateCourseInfo,
+            int customisationId,
+            int centreId,
+            int categoryId = 0
+        )
+        {
+            var result = GetCourseCustomPromptsResultForCourse(customisationId, centreId, categoryId);
+
+            return PopulateCustomPromptWithAnswerListFromCourseAdminFieldsResult(result, delegateCourseInfo);
+        }
+
+        public void UpdateCustomPromptForCourse(int customisationId, int promptNumber, bool mandatory, string? options)
+        {
+            courseAdminFieldsDataService.UpdateCustomPromptForCourse(customisationId, promptNumber, mandatory, options);
+        }
+
+        public void RemoveCustomPromptFromCourse(int customisationId, int promptNumber)
+        {
+            using var transaction = new TransactionScope();
+            try
+            {
+                userDataService.DeleteAllAnswersForAdminField(customisationId, promptNumber);
```

Comment:
This feels like something that should live next to other Progress/course related bits (or in courseAdminFieldsDataService) rather than in userDataService (as it's unrelated to Candidates/AdminUsers tables)

Reply:
I agree, I've moved this into courseAdminFieldsDataService